server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1843,7 +1843,7 @@ public CacheHelper getReaderCacheHelper() {
 
     }
 
-    public void close(String reason, boolean flushEngine) throws IOException {
+    public void close(String reason, boolean flushEngine, boolean deleted) throws IOException {
```

Comment:
Will overriding `close` to pass the `deleted` parameter help?

Reply:
Since there is only one consumer of this except tests, i am not  overloading  this function .